client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
+**Note**: Please make sure that your step has a unique slug.
 
-There is a required `isSignupFlow` flag that _MUST be `true` for signup flows_ (generally where a new site may be created), and should be `false` for other flows. The `isSignupFlow` flag controls whether we'll trigger a `calypso_signup_start` Tracks event when the flow starts. For signup flows, you can also supply additional event props to the `calypso_signup_start` event by implementing the optional `useTracksEventProps()` hook on the flow.
+**A step is simply a React component that**:
+
+1. Renders UI that collects user input.
+2. Submit said input to the flow.
+3. May or may not accept props from the flow.
+
+#### Code example
 
 ```tsx
 /**
- * This is the return type of useStepNavigation hook
+ * Each step must be typed as Step. And it should declare the types of the data it submits and the data it accepts.
  */
-export type NavigationControls = {
+const SelectImportedSiteSource: Step< {
+	// This steps submits `platform` and `url` strings.
+	submits: {
+		platform: 'Wix' | 'Squarespace';
+		url: string;
+	};
+	// And it accepts the following props. Adding them here will make them available in the `props` object.
+	accepts: {
+		title?: string;
+		subTitle?: string;
+	};
+} > = function ImportStep( props ) {
+	const siteSlug = useSiteSlug();
 	/**
-	 * Submits the answers provided in the flow
+	 * `navigation` prop is provided by Stepper. It allows your step to submit.
 	 */
-	submit?: ( providedDependencies?: ProvidedDependencies, ...params: string[] ) => void;
+	const { navigation, title, subTitle } = props;
+
+	return (
+		<div>
+			<h1>{ title || 'Welcome to WordPress.com' }</h1>
+			<h2>{ subTitle || 'Time to migrate you a site!' }</h2>
+
+			<form
+				onSubmit={ ( event ) =>
+					navigation.submit( {
+						platform: event.currentTarget.platform,
+						url: event.currentTarget.url,
+					} )
+				}
+			>
+				<label>
+					Enter the URL of your site
+					<input type="url" name="url" />
+				</label>
+
+				<label>
+					Pick your current site platform
+					<select name="platform">
+						<option value="wix">Wix</option>
+						<option value="ss">Squarespace</option>
+					</select>
+				</label>
+
+				<button>Submit</button>
+			</form>
+		</div>
+	);
 };
+
+export default SelectImportedSiteSource;
 ```
 
-Since this is a hook, it can access any state from any store, so you can make dynamic navigation decisions based on the state. [Here](./declarative-flow/site-setup-flow.ts) is a developed example of this hook.
+#### Passing data down to steps
+
+It is often the case that you want to customize steps around your flow. You can do that via `useStepsProps`.
+
+#### Reusability
+
+Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of the flows, flows just happen to use them.
```

Comment:
```suggestion
Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of a flow, flows just happen to use them.
```